logstash-core/lib/logstash/util/wrapped_acked_queue.rb

```diff
@@ -287,6 +288,12 @@ def cancel(event)
         # @cancelled[event] = true
       end
 
+      def collect
```

Comment:
Two comments -- in Ruby, `Enumerable#collect` returns an Enumerator, not an array.

If you wish to produce an array with an object implementing `each` you can `include Enumerable`:

```
% irb
>> class X
>>   include Enumerable
>>   def each(&block)
>>     block.call(1)
>>     block.call(2)
>>     block.call(3)
>>   end
>> end
=> nil
>> X.new.to_a
=> [1, 2, 3]
```

In this case, removing this `collect` method and adding `include Enumerable` will give you a `to_a` method to get an array with all elements of the `#each` iteration.

Reply:
@jordansissel I think there was some performance bug with that approach out there. Even without a bug, this will add another frame and slow things down. Better not add more indirection to the hot-path here :)

Reply:
Also there's this one https://github.com/jruby/jruby/issues/4108 which makes me less than confident in using `Enumerable` at this point :D

Reply:
Thanks for the explanation @original-brownbear . I'm +1 on this approach given your concerns.